adc/ad7779/ad7779.c

```diff
@@ -632,55 +672,95 @@ static int ad7779_gpio_init(void)
 
 	/* Configure as GPIOs */
 
-	/* Leave /START alone, it is pulled up, as it should be */
-
+	/* /START */
+	pctl.iomux.mux = pctl_mux_gpio_b0_04;
+	platformctl(&pctl);
 	/* /RESET */
 	pctl.iomux.mux = pctl_mux_gpio_b0_05;
 	platformctl(&pctl);
 	/* /CLK_SEL */
 	pctl.iomux.mux = pctl_mux_gpio_b0_06;
 	platformctl(&pctl);
+	/* Hardware reset */
+	pctl.iomux.mux = pctl_mux_gpio_emc_23;
+	platformctl(&pctl);
 
 	pctl.type = pctl_iopad;
 	pctl.iopad.hys = 0;
-	pctl.iopad.pus = 0;
+	pctl.iopad.pus = 0b11;
 	pctl.iopad.pue = 0;
 	pctl.iopad.pke = 1;
 	pctl.iopad.ode = 0;
 	pctl.iopad.speed = 2;
-	pctl.iopad.dse = 6;
-	pctl.iopad.dse = 3;
+	pctl.iopad.dse = 1;
+	pctl.iopad.sre = 0;
 
+	/* /START */
+	pctl.iopad.pad = pctl_pad_gpio_b0_04;
+	platformctl(&pctl);
 	/* /RESET */
 	pctl.iopad.pad = pctl_pad_gpio_b0_05;
 	platformctl(&pctl);
 	/* /CLK_SEL */
 	pctl.iopad.pad = pctl_pad_gpio_b0_06;
 	platformctl(&pctl);
+	/* Hardware reset */
+	pctl.iopad.pad = pctl_pad_gpio_emc_23;
+	platformctl(&pctl);
 
 	/* Set states */
+	gpio_setDir(id_gpio2, 4, 1);
 	gpio_setDir(id_gpio2, 5, 1);
 	gpio_setDir(id_gpio2, 6, 1);
+	gpio_setDir(id_gpio4, 23, 1);
 
+	gpio_setPin(id_gpio2, 4, 1);
 	gpio_setPin(id_gpio2, 5, 1);
 	gpio_setPin(id_gpio2, 6, 1);
+	gpio_setPin(id_gpio4, 23, 0);
 
 	return AD7779_OK;
 }
 
 
-static int ad7779_reset(void)
+static int ad7779_reset(int hard)
 {
+	int iter = 0;
+	uint8_t status[17];
+
+	/* Hardware reset */
+	if (hard) {
+		gpio_setPin(id_gpio4, 23, 1);
+		usleep(200000);
+		gpio_setPin(id_gpio4, 23, 0);
+	}
+
+	/* Software reset */
+	gpio_setPin(id_gpio2, 4, 0);
+	usleep(10000);
 	gpio_setPin(id_gpio2, 5, 0);
+	usleep(200000);
+	gpio_setPin(id_gpio2, 4, 1);
 	usleep(100000);
 	gpio_setPin(id_gpio2, 5, 1);
-	usleep(100000);
 
-	return AD7779_OK;
+	memset(status, 0, sizeof(status));
+
+	do {
+		if (!ad7779_get_status(status)) {
+			if (status[16] & 0x10)
+				return AD7779_OK;
+		}
+
+		++iter;
+		usleep(100000);
+	} while (iter < 4);
```

Comment:
Wouldn't `for` loop be better here?

Reply:
Right, for loop would do better. Will fix that.

Reply:
Done